ocs_ci/ocs/resources/storage_cluster.py

```diff
@@ -498,3 +499,118 @@ def get_all_storageclass():
+                    old_osd_flag = 1
+                    sum_of_percentage_old_osd_first_utilization += first_utilization[osd_name]
+                    sum_of_percentage_old_osd_second_utilization += second_utilization[osd_name]
+            if old_osd_flag == 0:
+                sum_of_percentage_new_osd_first_utilization += first_utilization[osd_name]
+                sum_of_percentage_new_osd_second_utilization += second_utilization[osd_name]
+        if sum_of_percentage_old_osd_first_utilization < sum_of_percentage_old_osd_second_utilization:
+            log.info(f"sum of old osd utilization is not decreasing"
+                     f" first util sum is: {sum_of_percentage_old_osd_first_utilization}"
+                     f" and second util sum is "
+                     f"{sum_of_percentage_old_osd_second_utilization}")
+            num_of_fails_old_osd_increase += 1
+        else:
+            log.info(f"sum of old osd is decreasing from {sum_of_percentage_old_osd_first_utilization} "
+                     f"to {sum_of_percentage_old_osd_second_utilization}")
+        if sum_of_percentage_new_osd_first_utilization > sum_of_percentage_new_osd_second_utilization:
+            log.info(f"sum of new osd utilization is not increasing"
+                     f" first util sum is: {sum_of_percentage_new_osd_first_utilization}"
+                     f" and second util sum is "
+                     f"{sum_of_percentage_new_osd_second_utilization}")
+            num_of_fails_new_osd_decrease += 1
+        else:
+            log.info(f"sum of new osd is increasing from {sum_of_percentage_new_osd_first_utilization} "
+                     f"to {sum_of_percentage_new_osd_second_utilization}")
+        if num_of_fails_old_osd_increase > 1 or num_of_fails_new_osd_decrease > 1:
+            log.error(f"Cluster failed to balance more than 2 times out of 3")
+            raise ClusterUtilizationNotBalanced(f"Cluster failed to balance more than 2 times out of 3")
+
+
+@retry(OsdIsIncreasing, tries=10, delay=60, backoff=1)
+def check_until_osd_ratio_start_decrease_or_equal(osds, digit_point=None, ratio_stable=None):
+    """
+    The function gets a list of OSDs and check if the sum of utilization percentage is decreasing or stay equal
+    depends on what was requested. It checks for 3 times in a row and tolerates one fail out of 3.
+    Args:
+        osds (list): List of osds ie [osd.1, osd.2...]
+        digit_point (int): How many digit after decimal point to check. It was created because monitoring data
+        is always added so to cancel the increase of monitoring data you can choose like 2
+        ratio_stable(anything): If not none it will check that in time of 3 rounds the osd is not increasing and
+        can tolerate 1 fail out of 3
+
+    """
+    from time import sleep
+    from ocs_ci.ocs.cluster import get_osd_utilization
+
+    log.info(f"Starting 3 loop to check utilization")
+    for t in range(1, 4):
```

Comment:
again, try to use `while` instead, with an exit condition

Reply:
the retry of the whole function have exit condition. I want to check that 2 times out of 3 the increase and decrease of data is happening. If you mean a loop with a  variable that is increased with +1 and then an exit condition according to that I don't understand the reason.

Reply:
Ok. I understand. The exit condition will be 2 out of 3 :)

Reply:
by exit condition, I meant something like a timeout.